base/base/scope_guard.h

```diff
@@ -29,11 +29,13 @@ class [[nodiscard]] BasicScopeGuard
     requires std::is_convertible_v<G, F>
     constexpr BasicScopeGuard & operator=(BasicScopeGuard<G> && src) // NOLINT(cppcoreguidelines-rvalue-reference-param-not-moved, cppcoreguidelines-noexcept-move-operations)
     {
-        if (this != &src)
+        if constexpr (std::is_same_v<G, F>)
```

Comment:
I found this issue when trying to make `scope_guard` work without `std::optional`. In the end I didn't need this change, but I thought I will leave it in as it is a good fix.